### PR TITLE
Blocks should be synchronized on read-only fields

### PR DESCRIPTION
**Pull request by marco-carvalho**
Continuation of https://github.com/Ryujinx/Ryujinx/pull/5212

### CODE REVIEW

File: src/Ryujinx.Graphics.GAL/Multithreading/BufferMap.cs [hunk 1]
```diff
@@ -15,9 +15,9 @@ class BufferMap
     {
         private ulong _bufferHandle = 0;
 
-        private Dictionary<BufferHandle, BufferHandle> _bufferMap = new Dictionary<BufferHandle, BufferHandle>();
-        private HashSet<BufferHandle> _inFlight = new HashSet<BufferHandle>();
-        private AutoResetEvent _inFlightChanged = new AutoResetEvent(false);
+        private readonly Dictionary<BufferHandle, BufferHandle> _bufferMap = new Dictionary<BufferHandle, BufferHandle>();
+        private readonly HashSet<BufferHandle> _inFlight = new HashSet<BufferHandle>();
+        private readonly AutoResetEvent _inFlightChanged = new AutoResetEvent(false);
```
Comment (AcK77) [inline src/Ryujinx.Graphics.GAL/Multithreading/BufferMap.cs:20]:
```suggestion
        private readonly Dictionary<BufferHandle, BufferHandle> _bufferMap = new();
        private readonly HashSet<BufferHandle> _inFlight = new();
        private readonly AutoResetEvent _inFlightChanged = new(false);
```

File: src/Ryujinx.Graphics.GAL/Multithreading/SyncMap.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ namespace Ryujinx.Graphics.GAL.Multithreading
 {
     class SyncMap : IDisposable
     {
-        private HashSet<ulong> _inFlight = new HashSet<ulong>();
+        private readonly HashSet<ulong> _inFlight = new HashSet<ulong>();
```
Comment (AcK77) [inline src/Ryujinx.Graphics.GAL/Multithreading/SyncMap.cs:9]:
```suggestion
        private readonly HashSet<ulong> _inFlight = new();
```

File: src/Ryujinx.Graphics.Gpu/Image/Texture.cs [hunk 3]
```diff
@@ -174,7 +174,7 @@ private struct TexturePoolOwner
         public bool HasViews => _views.Count > 0;
 
         private int _referenceCount;
-        private List<TexturePoolOwner> _poolOwners;
+        private readonly List<TexturePoolOwner> _poolOwners = new List<TexturePoolOwner>();
```
Comment (AcK77) [inline src/Ryujinx.Graphics.Gpu/Image/Texture.cs:177]:
```suggestion
        private readonly List<TexturePoolOwner> _poolOwners = new();
```

File: src/Ryujinx.Graphics.OpenGL/Sync.cs [hunk 4]
```diff
@@ -17,7 +17,7 @@ private class SyncHandle
         private ulong _firstHandle = 0;
         private ClientWaitSyncFlags _syncFlags => HwCapabilities.RequiresSyncFlush ? ClientWaitSyncFlags.None : ClientWaitSyncFlags.SyncFlushCommandsBit;
 
-        private List<SyncHandle> _handles = new List<SyncHandle>();
+        private readonly List<SyncHandle> _handles = new List<SyncHandle>();
```
Comment (AcK77) [inline src/Ryujinx.Graphics.OpenGL/Sync.cs:20]:
```suggestion
        private readonly List<SyncHandle> _handles = new();
```

File: src/Ryujinx.HLE/HOS/Services/Mm/IRequest.cs [hunk 5]
```diff
@@ -7,7 +7,7 @@ namespace Ryujinx.HLE.HOS.Services.Mm
     [Service("mm:u")]
     class IRequest : IpcService
     {
-        private static object                  _sessionListLock = new object();
+        private static readonly object _sessionListLock = new();
         private static List<MultiMediaSession> _sessionList     = new List<MultiMediaSession>();
```
Comment (AcK77) [inline src/Ryujinx.HLE/HOS/Services/Mm/IRequest.cs:11]:
```suggestion
        private static List<MultiMediaSession> _sessionList = new();
```

**Review by AcK77 — CHANGES_REQUESTED** [review]

**Review by AcK77 — APPROVED** [review]

**Review by marysaka — APPROVED** [review]

**Comment by marco-carvalho** [comment]
@marysaka can I ask why this is now closed, since you and @AcK77 previously approved it?

**Comment by marysaka** [comment]
> @marysaka can I ask why this is now closed, since you and @AcK77 previously approved it?

Already part of https://github.com/Ryujinx/Ryujinx/pull/5380 planned for merge soon.

Also, I do not believe that flooding PRs from sonar lint or other tools is useful for the project.
We barely have time to review actual changes in the codebase and this just end up slowing us down more than anything.

**Pull request closed without merging**